.github/workflows/integration_tests.yml

```diff
@@ -49,12 +57,13 @@ jobs:
           python -m pip install --upgrade pip
           pip install . psrecord
       - name: Run and profile workflow
+        env:
+          branch_ref: ${{ steps.branch_ref.outputs.branch_ref }}
         run: |
-          branch_ref="${{ github.ref }}"
-              if [ -z "$branch_ref" ]; then
-                echo "Error: $branch_ref is empty. This workflow requires a valid branch reference."
-                exit 1
-              fi
+          if [ -z "$branch_ref" ]; then
+            echo "Error: $branch_ref is empty. This workflow requires a valid branch reference."
+            exit 1
+          fi
           if [[ $branch_ref == 'refs/heads/master' ]]; then
```

Comment:
Because `branch_ref` is fully formed  in the format `refs/heads/<name of the branch>` this line should evaluate to true when it is run on the `master` branch and we should get the profile runs in the `master` branch.